File: app/src/main/java/ru/vpiska/shop/ShopApp.java
package ru.vpiska.shop;

import java.util.Comparator;

/**
 * Created by Кирилл on 22.12.2018.
 */

public class ShopApp {

    private String productId,title;
    double money;



    ShopApp(String productId,String title,double price){
        this.productId = productId;
        this.title = title;
        this.money = Math.round(price);
    }

    public static Comparator<ShopApp> ShopComparator = new Comparator<ShopApp>() {

        public int compare(ShopApp s1, ShopApp s2) {


            //ascending order
            return Double.compare(s1.getPrice(), s2.getPrice());

            //descending order
            //return StudentName2.compareTo(StudentName1);
        }};

    public String getProductId() {
        return productId;
    }

    public String getTitle() {
        return title;
    }

    public double getPrice() {
        return money;
    }
}

File: app/src/main/java/ru/vpiska/shop/ShopApp.kt
package ru.vpiska.shop

import java.util.*

/**
 * Created by Кирилл on 22.12.2018.
 */

class ShopApp internal constructor(val productId: String, val title: String, price: Double) {
    var price: Double = 0.toDouble()
        internal set


    init {
        this.price = Math.round(price).toDouble()
    }

    companion object {

        var ShopComparator: Comparator<ShopApp> = Comparator { s1, s2 ->
            //ascending order
            java.lang.Double.compare(s1.price, s2.price)

            //descending order
            //return StudentName2.compareTo(StudentName1);
        }
    }
}
